core/sail/api/src/main/java/org/eclipse/rdf4j/common/concurrent/locks/ReadPrefReadWriteLockManager.java

```diff
@@ -69,12 +70,7 @@ public Lock tryReadLock() {
 	 * released.
 	 */
 	@Override
-	public synchronized Lock getReadLock() throws InterruptedException {
-		// Wait for the writer to finish
-		while (isWriterActive()) {
-			waitForActiveWriter();
-		}
-
+	public Lock getReadLock() throws InterruptedException {
```

Comment:
I'm not sure you can get rid of synchronization here. The `createReadLock` method itself is not thread-safe and its implementation has a possible race condition with checking the state of the write lock. Callers of `createReadLock` should take care of synchronization I think...

Reply:
Previous code spins while waiting for active writers to complete, that is now handled by the underlying code instead.